x-pack/plugins/ml/common/types/anomaly_detection_jobs/job.ts

```diff
@@ -13,6 +13,7 @@ export type BucketSpan = string;
 export interface CustomSettings {
   custom_urls?: UrlConfig[];
   created_by?: CREATED_BY_LABEL;
+  job_tags?: Record<string, string>;
```

Comment:
nit, we try to only use Record when the key is known. e.g.
`Record<'key1' | 'key2', string>` 

for data where the key will be user defined, we prefer using a plain index signature.
```
job_tags?: {
  [tag: string]: string
}
```